pycfmodel/model/resources/properties/statement.py

```diff
@@ -53,9 +53,10 @@ class Statement(Property):
     Condition: Optional[StatementCondition] = None
 
     @validator("Effect")
-    def capitalize_if_str(cls, v: ResolvableStr):
-        if isinstance(v, str):
-            return v.capitalize()
+    def allowed_values_for_effect_and_capitalized(cls, v: ResolvableStr):
+        v = v.capitalize()
+        if v not in ["Allow", "Deny"]:
+            raise ValueError("Effect on Statement must either be Allow or Deny")
```

Comment:
Check if there is a functiondict or string. With a funcitondict this would break

Reply:
I believe what you meant is what I did in [90cdeea](https://github.com/Skyscanner/pycfmodel/pull/101/commits/90cdeea3911efa00bc2d7c1718f18fc9ff316555)